stl/src/tzdb.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <internal_shared.h>
+#include <xtzdb.h>
+
+#include <Windows.h>
+
+#pragma comment(lib, "Advapi32")
+
+_EXTERN_C
+
+_RegistryLeapSecondInfo* __stdcall __std_tzdb_get_reg_leap_seconds(
+    const size_t prev_reg_ls_size, size_t* current_reg_ls_size) {
+    // On exit---
+    //    *current_reg_ls_size <= prev_reg_ls_size, *reg_ls_data == nullptr --> no new data
+    //    *current_reg_ls_size >  prev_reg_ls_size, *reg_ls_data != nullptr --> new data, successfully read
+    //    *current_reg_ls_size == 0,                *reg_ls_data != nullptr --> new data, failed reading
+    //    *current_reg_ls_size >  prev_reg_ls_size, *reg_ls_data == nullptr --> new data, failed allocation
+
+    constexpr auto reg_key_name    = TEXT("SYSTEM\\CurrentControlSet\\Control\\LeapSecondInformation");
+    constexpr auto reg_subkey_name = TEXT("LeapSeconds");
+    *current_reg_ls_size           = 0;
+    HKEY leap_sec_key              = 0;
+
+    LSTATUS status = RegOpenKeyEx(HKEY_LOCAL_MACHINE, reg_key_name, 0, KEY_READ, &leap_sec_key);
+    if (status != ERROR_SUCCESS) {
+        // May not exist on older systems. Treat this as equivalent to the key existing but with no data.
+        return nullptr;
+    }
+
+    DWORD byte_size = 0;
+    status          = RegQueryValueEx(leap_sec_key, reg_subkey_name, nullptr, nullptr, nullptr, &byte_size);
+    static_assert(sizeof(_RegistryLeapSecondInfo) == 12);
+    const auto ls_size   = byte_size / 12;
+    *current_reg_ls_size = ls_size;
+
+    _RegistryLeapSecondInfo* reg_ls_data = nullptr;
+    if ((status == ERROR_SUCCESS || status == ERROR_MORE_DATA) && ls_size > prev_reg_ls_size) {
+        try {
+            reg_ls_data = new _RegistryLeapSecondInfo[ls_size];
+            status      = RegQueryValueEx(
+                leap_sec_key, reg_subkey_name, nullptr, nullptr, reinterpret_cast<LPBYTE>(reg_ls_data), &byte_size);
+            if (status != ERROR_SUCCESS) {
+                *current_reg_ls_size = 0;
+            }
+        } catch (...) {
+        }
+    }
+
+    RegCloseKey(leap_sec_key);
+    if (status != ERROR_SUCCESS) {
+        SetLastError(status);
+    }
+
+    return reg_ls_data;
+}
+
+void __stdcall __std_decalloc_reg_leap_seconds(_RegistryLeapSecondInfo* _Rlsi) {
+    delete[] _Rlsi;

```

Comment:
The name `decalloc` doesn't correspond to the `new[]` / `delete[]` that's being used. I believe that the allocation/deallocation machinery being used is fine (being within `tzdb.cpp` avoids any mismatch, and the allocation is short-lived within a `<chrono>` function call, so it can be a "normal" block instead of a CRT block), so only the function name seems inconsistent (possibly a relic of development?).

@d-winsor notes that the `tzdb` prefix would be valuable here. As this is paired with `__std_tzdb_get_reg_leap_seconds`, I would recommend `__std_tzdb_delete_reg_leap_seconds`.